core/validatorapi/validatorapi_test.go

```diff
@@ -1321,3 +1321,39 @@ func TestComponent_SubmitAggregateAttestations(t *testing.T) {
 
 	require.NoError(t, vapi.SubmitAggregateAttestations(ctx, []*eth2p0.SignedAggregateAndProof{agg}))
 }
+
+func TestComponent_SubmitSyncCommitteeMessages(t *testing.T) {
+	const vIdx = 1
+
+	var (
+		ctx = context.Background()
+		msg = &altair.SyncCommitteeMessage{
+			Slot:            testutil.RandomSlot(),
+			BeaconBlockRoot: testutil.RandomRoot(),
+			ValidatorIndex:  vIdx,
+			Signature:       testutil.RandomEth2Signature(),
+		}
+		pubkey = beaconmock.ValidatorSetA[vIdx].Validator.PublicKey
+	)
+
+	bmock, err := beaconmock.New(beaconmock.WithValidatorSet(beaconmock.ValidatorSetA))
+	require.NoError(t, err)
+
+	vapi, err := validatorapi.NewComponentInsecure(t, bmock, 0)
+	require.NoError(t, err)
+
+	vapi.Subscribe(func(_ context.Context, duty core.Duty, set core.ParSignedDataSet) error {
+		require.Equal(t, core.NewSyncMessageDuty(int64(msg.Slot)), duty)
+
+		pk, err := core.PubKeyFromBytes(pubkey[:])
+		require.NoError(t, err)
+
+		data, ok := set[pk]
+		require.True(t, ok)
+		require.Equal(t, core.NewPartialSignedSyncMessage(msg, 0), data)
```

Comment:
how do you know this code is actually called? (please do not use ctx cancel pattern, anymore, rather use something like

```
done := make(chan struct)
...
close(done)
...
<-done   
```

Reply:
we can add a counter which is incremented each time the subscription function is called. we can use it to assert that this code block is actually called.

Reply:
Sure, but this test doesn't use the ctx cancel pattern

Reply:
as long as it is threadsafe then sure